### ISSUE TITLE
Support JSON

### ISSUE THREAD

**Issue by allejo** [BZFlag-Dev/bzflag#7]
As per my discussion with blast, it would be nice to support a modern data format such as JSON at least in the API to replace the current .ini format that's very limited. I would like to suggest the [json-c](https://github.com/json-c/json-c) library, which is what I use in my League Overseer plugin, since it has a package on most Linux distributions as `libjson0-dev` or `json-c-devel` and it has support for Windows as well. Which library we choose is also up for discussion.

We would be able to use JSON internally for configuration files and other files as well. When to implement this whether for 2.4.x or 2.5.x is up for discussion.

/cc @blast007

**Comment by jwmelto** [comment]
JSON is an abomination of a data format... it's a giant security hole for JavaScript (where it was invented) and Python, and not particularly pleasant for C++. I don't see any advantage to using it here.

**Comment by kongr45gpen** [comment]
I think [YA](http://yaml.org/)[ML](http://en.wikipedia.org/wiki/YAML) would be a much better option to use for configuration files with JSON, since the latter doesn't support comments and is a bit hard to read and edit.

**Comment by blast007** [comment]
If we're adding such a library, it should ideally be something that can be used not just for configuration files but data exchanges over HTTP, and should be something with wide language support (at least C/C++, Python, PHP, Perl).  YAML might fit that, but I'd have to see how nice it is for something like the server list.

**Comment by allejo** [comment]
I agree with alezakos that YAML would be ideal for configuration files since they're far easier to read and can have comments, but I'm not sure how well YAML can be used for returning data for HTTP communication. I don't think I've ever seen a YAML REST API, I've always seen XML or JSON as the de facto.

**Comment by kongr45gpen** [comment]
[You can actually use a YAML parser to read most JSON documents](http://yaml.org/spec/1.2/spec.html#id2759572), although a JSON parser might be faster for that task.

**Comment by bz-next** [comment]
nlohmann json (https://github.com/nlohmann/json) can be integrated as a single header library and only requires C++11. I've used it a few times and it is very easy to integrate. My experimental fork uses it (very crudely).

I'd suggest JSON for web-facing things and then decide on a configuration format independent of that (could also be JSON, could be YAML, could be something else.)
